### PR TITLE
Update Jsonschema package to V4

### PR DESCRIPTION
**Pull request by rohithvarma3000**
### :pencil: Description

**Type:** :beetle: `bugfix`

Write a complete description of your changes, including the necessary context or any piece of information required to understand your work.

Also, link issues affected by this pull request by using the keywords: `close`, `closes`, `closed`, `fix`, `fixes`, `fixed`, `resolve`, `resolves` or `resolved`. 

Attempt to update the jsonschema package to v4 and handle the new bugs introduced. [Link](https://github.com/tardis-sn/tardis/issues/1807) to issue.

### :pushpin: Resources

Examples, notebooks, and links to useful references.
https://python-jsonschema.readthedocs.io/en/v3.2.0/validate/#the-validator-interface

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by rohithvarma3000** [comment]
I have put the extra yaml file for now so that it can be used to test the PR. I will remove it before the PR merges

**Comment by andrewfullard** [comment]
Good work on the validator. However, the environment file should be merged with the existing one unless somebody told you otherwise

File: tardis/io/config_validator.py [hunk 1]
```diff
@@ -62,6 +62,10 @@ def _yaml_handler(path):
         return yaml.load(f, Loader=YAMLLoader)
 
 
+def is_Quantity(checker, instance):
+    return isinstance(instance, Quantity)
```
Comment (andrewfullard) [inline tardis/io/config_validator.py:66]: Add a short docstring

**Review by andrewfullard — CHANGES_REQUESTED** [review]

**Comment by epassaro** [comment]
@rohithvarma3000 You also need to render both lockfiles by installing `conda-lock` and doing `conda lock render conda-lock.yml`.

Push both lockfiles to this pull request and see if tests pass. As you can see, the [`tests.yml`](https://github.com/tardis-sn/tardis/blob/master/.github/workflows/tests.yml#L68) workflow runs tests in an environment made with the lockfiles.

File: tardis/io/config_validator.py (hunk 1 above)
Comment (rohithvarma3000) [reply tardis/io/config_validator.py:82]: Added the docstring.

**Comment by rohithvarma3000** [comment]
@epassaro I have regenerated the lockfiles and pushed them to the branch.

**Comment by andrewfullard** [comment]
The lockfile has had a `win-64` platform added. Was this intentional? We don't officially support Windows right now. @epassaro this is potentially a problem.

**Comment by epassaro** [comment]
> The lockfile has had a `win-64` platform added. Was this intentional? We don't officially support Windows right now. @epassaro this is potentially a problem.

Hi @rohithvarma3000, check the `pre-release.yml` workflow and look for the correct command to lock files only for linux/osx and then render again

**Comment by rohithvarma3000** [comment]
Thanks for the suggestion @epassaro. I have made the necessary changes to the PR.

**Review by epassaro — APPROVED** [review]

**Comment by epassaro** [comment]
This looks good to me @andrewfullard , I approved the pull request.

**Pull request merged**